helpers/error.py

```diff
@@ -18,6 +19,28 @@ class Error(Exception):
     pass
 
 
+@dataclass
+class MemoryConsumptionError(Error):
```

Comment:
`Error` base class would confuse everyone. (I understand that it was here already) The class `Error` doing nothing and its implementation has no sense (maybe to have custom base class). I would remove `Error` class at all, or, at least, if you wan to have custom base class for all custom exceptions, please, rename it  to something like `FrameworkBaseException`.   And I would rename new Exception to have `Exception` suffix not `Error`

Reply:
I will fix this with a separate PR when I replace dbg on logging library.